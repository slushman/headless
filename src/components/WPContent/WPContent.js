import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import EmbedContainer from 'react-oembed-container';

const Content = styled.div`
	& a {
		--link-color: var(--color-blue);
		
		color: var(--link-color);
	}

	& a:visited,
	& a:active,
	& a:hover {
		--link-color: var(--color-blue);
	}

	& a:focus {
		outline-color: #E8671C;
	}

	& p {
		line-height: 1.75;
		margin: 0 0 1.5em;
	}

	& pre,
	& code,
	& kbd,
	& samp, {
		font-family: 'Source Code Pro', monospace;
		position: relative;
	}

	& pre {
		color: var(--color-light);
		padding: 1.5em 0;
		position: relative;
		z-index: 1;
	}

	& pre:before {
		background-color: var(--color-dark);
		bottom: 0;
		content: "";
		left: 50%;
		margin-right: -50vw;
		margin-left: -50vw;
		position: absolute;
		right: 50%;
		top: 0;
		width: 100vw;
		z-index: -1;
	}

	& pre code {
		display: block;
		font-family: 'Source Code Pro', monospace;
		font-size: 1rem;
		overflow: scroll;
		padding: 1em;
	}
`;

const WPContent = ({content}) => {
	return (
		<EmbedContainer markup={content.rendered}>
			<Content dangerouslySetInnerHTML={{ __html: content.rendered }} />
		</EmbedContainer>
	);
};

WPContent.propTypes = {
	content: PropTypes.object
};

export default WPContent;