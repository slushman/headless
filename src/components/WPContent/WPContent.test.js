import React from 'react';
import renderer from 'react-test-renderer';

import WPContent from './WPContent';

const content = {
	rendered: "Testing simulated content from WordPress."
};

it('renders without crashing', () => {
	const component = renderer.create(
		<WPContent content={content} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
