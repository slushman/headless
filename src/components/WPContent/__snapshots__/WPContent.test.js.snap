// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className={null }
>
  <div
    className="sc-bdVaJa ebydHU"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Testing simulated content from WordPress.",
      }
    }
  />
</div>
`;
